src/Login/Login.js

```diff
@@ -48,8 +50,38 @@ const Login = ({navigation}) => {
                 })
             }
         } else {
-            firebase.auth().signInWithEmailAndPassword(username, password).then(function(user) {
-                var user = firebase.auth().currentUser;
+            firebase.auth().setPersistence(firebase.auth.Auth.Persistence.LOCAL) // persistent login can only be cleared on user sign out
+                .then(function() {                
+                    firebase.auth().signInWithEmailAndPassword(username, password)
+                        .then(function(user) {
+                            const currentUser = firebase.auth().currentUser;
+                            const db = firebase.database().ref('users').child(currentUser.uid);
+                            db.on('value', snap => {
+                                if (snap.val()) {
+                                    setContext({
+                                        userData: snap.val(),
+                                        userUid: currentUser.uid
+                                    })
+                                    navigation.navigate('Home', {screen: 'Timer'});
+                                } 
+                            }, error => alert(error))
+                    }, function (error){
+                        console.log(error.message);
+                        setErrorMessage(error.message);
+                    })
+            })
+            .catch(function(error) {
+                console.log(error.message);
+                setErrorMessage(error.message);
+            })
+        }    
+    }
+
+    // Checks if there is an existing user session
+    // If so, sets the user context and then navigates to home stack
+    const checkUserSession = () => {
```

Comment:
New function that looks for current user when login renders
If user found then set the user context and navigate to home screen